### DISCUSSION TITLE
Bisakah kita menyortir asset gambar/file berdasarkan tanggal uploadnya?

### DISCUSSION THREAD

**Discussion by lokersumatera** [mecha-cms/mecha#216] [Q&A] [answered]
Bisakah kita mensortir asset gambar/file berdasarkan tanggal uploadnya?

**Comment by taufik-nurrohman** [comment, marked as answer]
Bisa tapi belum tersedia API bawaannya. Paling manual pakai `asort()`.

~~~ php
$files = [];
foreach (g(LOT . D . 'asset', 'jpg') as $k => $v) {
    $files[$k] = filectime($k);
}

asort($files); // Sort by file creation time

foreach (array_keys($files) as $file) {
    echo $file . '<br>';
}
~~~

Panel versi berikutnya nanti bisa memungkinkan untuk mengeset _filter_, _recurse_, dan _chunk_ melalui URL.

https://github.com/mecha-cms/x.panel/commit/ae056a77748ea9ac9ba039c223009dea7946182f

**Contoh:**

 - `/panel/get/asset/1?deep=true&x=gif,jpeg,jpg,png` &rarr; untuk menampilkan semua gambar dalam satu laman aset secara rekursif.
 - `/panel/get/page/1?deep=true&query=lorem&x=draft` &rarr; untuk menampilkan semua draf laman yang mengandung kata kunci &ldquo;lorem&rdquo; secara rekursif.
